workitem.2.go

```diff
@@ -165,3 +168,77 @@ func buildAbsoluteURL(req *goa.RequestData) string {
 	}
 	return fmt.Sprintf("%s://%s%s", scheme, req.Host, req.URL.Path)
 }
+
+// ConvertWorkItemToJSONAPI is responsible for converting given WorkItem model object into a
+// response resource object by jsonapi.org specifications
+func (c *Workitem2Controller) ConvertWorkItemToJSONAPI(ctx *app.UpdateWorkitem2Context, wi *app.WorkItem) *app.WorkItem2 {
+	// construct default values from input WI
+
+	absoluteURL := buildAbsoluteURL(ctx.RequestData) // it includes path hence no modifications needed
+	op := &app.WorkItem2{
+		Links: &app.WorkItemLinks{
+			Self: &absoluteURL,
+		},
+		Data: &app.WorkItemDataForUpdate{
+			ID:   wi.ID,
+			Type: "workitems",
+			Attributes: map[string]interface{}{
+				"version": wi.Version,
+			},
+			Relationships: &app.WorkItemRelationships{
+				BaseType: &app.RelationshipBaseType{
+					Data: &app.BaseTypeData{
+						ID:   wi.Type,
+						Type: "workitemtypes",
+					},
+				},
+			},
+		},
+	}
+	// Move fields into Relationships or Attributes as needed
+	for name, val := range wi.Fields {
+		switch name {
+		case models.SystemAssignee:
+			if val != nil {
+				valStr := val.(string)
+				op.Data.Relationships.Assignee = &app.RelationAssignee{
+					Data: &app.AssigneeData{
+						ID:   &valStr,
+						Type: "identities",
+					},
+				}
+			}
+		default:
+			op.Data.Attributes[name] = val
+		}
+	}
+	return op
+}
+
+// Update does PATCH workitem
+func (c *Workitem2Controller) Update(ctx *app.UpdateWorkitem2Context) error {
+	return application.Transactional(c.db, func(appl application.Application) error {
+
+		toSave := app.WorkItemDataForUpdate{
+			ID:            ctx.Payload.Data.ID,
+			Type:          ctx.Payload.Data.Type,
+			Relationships: ctx.Payload.Data.Relationships,
+			Attributes:    ctx.Payload.Data.Attributes,
+		}
+		wi, err := appl.WorkItems2().Save(ctx, toSave)
+		if err != nil {
+			switch err := err.(type) {
```

Comment:
Please rewrite this switch like so to include the correct JSONAPI errors in the response.

```go
switch err := err.(type) {
case models.BadParameterError:
	jerrors, _ := jsonapi.ErrorToJSONAPIErrors(goa.ErrBadRequest(fmt.Sprintf("Error updating work item: %s", err.Error())))   
	return ctx.BadRequest(jerrors)
case models.NotFoundError:
	jerrors, _ := jsonapi.ErrorToJSONAPIErrors(goa.ErrNotFound(err.Error()))   
	return ctx.NotFound(jerrors)
case models.VersionConflictError:
	jerrors, _ := jsonapi.ErrorToJSONAPIErrors(goa.ErrBadRequest(fmt.Sprintf("Error updating work item: %s", err.Error())))   
	return ctx.BadRequest(jerrors)
default:
	log.Printf("Error updating work items: %s", err.Error())
jerrors, _ := jsonapi.ErrorToJSONAPIErrors(goa.ErrInternal(err.Error()))   
	return ctx.InternalServerError(jerrors)
}
```

This will only work if you update the `resources.go` and specify a `JSONAPIErrors` as the second parameter to the error response types:

```go
		a.Response(d.BadRequest, JSONAPIErrors)
		a.Response(d.InternalServerError, JSONAPIErrors)
		a.Response(d.NotFound, JSONAPIErrors)
```